lib/letsencrypt.js

```diff
@@ -1,28 +1,31 @@
 'use strict';
-const deepClone = require('deep-clone');
 
 function createServer(app, settings, certificates, leSettings, log) {
-    let server;
-
-    const leLog = function (debug, arg1, arg2, arg3) {
-        if (debug) {
-            if (arg1 && typeof arg1 === 'string' && arg1[0] === 'n' && arg1[1] === 'o') {
-                return;
+    return new Promise((resolve, reject) => {
+        if (!settings.secure) {
+            log.debug('Using http createServer');
+            resolve(require('http').createServer(app));
+        } else if (!settings.leEnabled) {
+            log.debug('Using https createServer');
+            resolve(require('https').createServer(certificates, app));
+        } else {
+            // prepare domains
+            // TODO: I presume this is here for backward compatibility so remove in some very future update.
+            if (typeof leSettings.domains === 'string') {
+                leSettings.domains = leSettings.domains.split(',');
+                for (let d = leSettings.domains.length - 1; d >= 0; d--) {
+                    leSettings.domains[d] = leSettings.domains[d].trim();
+                    // delete empty
+                    !leSettings.domains[d] && leSettings.domains.splice(d, 1);
+                }
             }
-            log.info(arg1 + (arg2 || '') + (arg3 || ''));
-        }
-    };
 
-    if (settings.secure) {
-        if (leSettings && !leSettings.email && settings.leEnabled) {
-            log.error('Please specify the email address and domains to use Let\'s Encrypt certificates!');
-        }
+            if (!leSettings || !leSettings.email || !Array.isArray(leSettings.domains) || leSettings.domains.length == 0) {
```

Comment:
please use `===` comparison

Reply:
Possibly a good idea, but having already checked it's an array I don't see the point.

Reply:
Doesn’t matter bf already changed, however we avoid == comparisons

Reply:
OK, fixed in https://github.com/ioBroker/ioBroker.js-controller/pull/1156/commits/8324cb1d4b8b7bfa3d382d0646698f5de359408f